### PR TITLE
Backfill name_versions.user_id from rss_log

### LINKED ISSUE #4252: Name footer Last modified shows wrong user
The "Last modified ..." line in the show Name footer shows the User who created the Name, instead of the User who last modified it.
Example: https://mushroomobserver.org/names/53423

I am the User who last modified this name. See https://mushroomobserver.org/activity_logs/268502

> Log for Puccinia circaeae Pers. (53423)
2026-05-13 05:02:16 PDT (-0700)	:	Name updated by Joe Cohen.
2016-04-07 20:48:54 PDT (-0700)	:	Approved by myxomop.
 
But the footer lists the User who created it:

> Last modified: 2026-05-13 05:02:16 PDT (-0700) by Danny Newman (myxomop)

### PR DESCRIPTION
**Pull request by JoeCohen**
> [!WARNING]  
> Deploy at a low-traffic hour. The website will be down while the migration is running.
> CC estimated that would take 5 minutes  locally, longer on the server.

## Summary

Backfills NameVersions UserId's from RssLog

- All `name_versions` rows created before PR #4257 hold the name **creator's** `user_id` rather than the editor's, because `update_name_version` set `ver.user_id` in memory but never called `ver.save` (fixed in #4257)
- `clone_versioned_model` (acts_as_versioned) copies the name's `user_id` column — the creator's id — into every new version row, so without the save the DB was never corrected
- This migration uses each name's rss_log, which always recorded the correct editor login via `user_add_with_date`, to backfill the right `user_id` into each `name_versions` row
- Each version row is matched to the closest rss_log entry by timestamp (both are second-precision; created in the same request, so they differ by ≤ 1 s); a 120 s window absorbs any slow-server edge cases

## Test plan

- [x] Run `bin/rails db:migrate` locally and verify name 53423 (Puccinia circaeae) footer now shows Joe Cohen as last modifier
- [x] Check a few other recently-edited names to confirm footer user is correct
- [x] Verify the migration is idempotent (re-running skips rows where `user_id` is already correct)
> After running the migration once:
> 1. Note the current user_id on a known-fixed version in the Rails console:
> `v = Name::Version.where(name_id: 53423).last`
> v.user_id  # should now be Joe Cohen's id
> 2. Re-run the migration's up method directly (Rails won't re-run it via db:migrate because it's already in schema_migrations):
`BackfillNameVersionUserIds.new.up`
> 3. Confirm the value didn't change:
`Name::Version.where(name_id: 53423).last.user_id ` # same id as before
>
> If you want to be more thorough — confirm no rows were touched — wrap step 2 in a transaction and add a counter
```ruby
require Rails.root.join("db/migrate/20260522000000_backfill_name_version_user_ids") ; ActiveRecord::Base.transaction do
  BackfillNameVersionUserIds.new.up
  raise ActiveRecord::Rollback  # leave DB untouched
end
```
> That lets you run the second pass safely without risk of any side effects, and the fact that it completes without error confirms the guard logic works.

Fixes #4252

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: db/migrate/20260522000000_backfill_name_version_user_ids.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    Name.includes(:rss_log).find_each do |name|
+      next unless name.rss_log
+
+      # parse_log returns [tag, args, time] newest-first; reverse for
+      # chronological order so we can reason about ordering.
+      entries = name.rss_log.parse_log.reverse
+      next if entries.empty?
+
+      versions = Name::Version.where(name_id: name.id).order(:version).to_a
```
[the remark this answers is not in the document]
Comment (JoeCohen) [reply db/migrate/20260522000000_backfill_name_version_user_ids.rb]: Copilot implemented in [6aebb69](https://github.com/MushroomObserver/mushroom-observer/pull/4322/commits/6aebb691690a729754e2fcbca29bc3c03bfde2fc)

**Comment by JoeCohen** [comment]
Closing. The backfill was accomplished by a rails runner script instead of the migration in this PR>
